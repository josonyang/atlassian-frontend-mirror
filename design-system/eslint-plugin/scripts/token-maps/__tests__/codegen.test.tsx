import { createSpacingStylesFromTemplate } from '../spacing-codegen-template';

describe('@atlaskit/eslint-plugin-design-system', () => {
	// spacing
	test('spacing styles are generated correctly', () => {
		expect(createSpacingStylesFromTemplate()).toMatchSnapshot();
	});
});
