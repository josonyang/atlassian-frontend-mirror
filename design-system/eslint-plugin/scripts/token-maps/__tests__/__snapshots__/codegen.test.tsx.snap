// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@atlaskit/eslint-plugin-design-system spacing styles are generated correctly 1`] = `
"export const positiveSpaceMap = {
    '0px': 'space.0',
    '2px': 'space.025',
    '4px': 'space.050',
    '6px': 'space.075',
    '8px': 'space.100',
    '12px': 'space.150',
    '16px': 'space.200',
    '20px': 'space.250',
    '24px': 'space.300',
    '32px': 'space.400',
    '40px': 'space.500',
    '48px': 'space.600',
    '64px': 'space.800',
    '80px': 'space.1000',
};
export type Space = keyof typeof positiveSpaceMap;

export const negativeSpaceMap = {
    '-2px': 'space.negative.025',
    '-4px': 'space.negative.050',
    '-6px': 'space.negative.075',
    '-8px': 'space.negative.100',
    '-12px': 'space.negative.150',
    '-16px': 'space.negative.200',
    '-20px': 'space.negative.250',
    '-24px': 'space.negative.300',
    '-32px': 'space.negative.400',
};
export type NegativeSpace = keyof typeof negativeSpaceMap;

export const allSpaceMap = { ...positiveSpaceMap, ...negativeSpaceMap };

export type AllSpace = keyof typeof allSpaceMap;
"
`;
